src/js/components/ServiceDetail.js

```diff
@@ -60,6 +59,15 @@ class ServiceDetail extends mixin(InternalStorageMixin, StoreMixin, TabsMixin) {
     });
   }
 
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+    this.checkForVolumes();
+  }
+
+  componentWillUpdate() {
```

Comment:
We're extending [React.Component](https://github.com/facebook/react/blob/master/src/isomorphic/modern/class/ReactComponent.js), which is the isomorphic base class of all react components and therefor doesn't provide any of the DOM related methods. The life-cycle methods you referred to, get called by `ReactCompositeComponent` if they are defined (e.g. [componentWillUpdate](https://github.com/facebook/react/blob/master/src/renderers/shared/stack/reconciler/ReactCompositeComponent.js#L445) or [componentDidMount](https://github.com/facebook/react/blob/master/src/renderers/shared/stack/reconciler/ReactCompositeComponent.js#L337)). 
We should therefore revisited our decision/guideline to always call `super` in all of the life-cycle methods, as these might be undefined (depending on your parent class), which throws errors and fails our [tests](https://github.com/dcos/dcos-ui/pull/443#discussion_r67486705) in some cases. Let's add this to our agenda and discuss it in one of our next meetings.